Track return tax refund correctly using return_request's request_items

rajrishav0711 opened the pull request:
[Jira](https://dekeo.atlassian.net/browse/ALPHA-332)

For blended partial returns, we currently track the tax refund in avalara for the items sent  by the customer to the winner supplier, though we refund a bigger amount to customer. So we need to look at return_request’s request_items instead of return_items while calculating.

Partial return with customer getting $45.78 refund:

<img width="1440" alt="Screen Shot 2019-09-17 at 5 00 13 PM" src="https://user-images.githubusercontent.com/9412986/65038716-375d4480-d96e-11e9-9234-d6bcc8e4c88e.png">

Avalara Entry for above return refund:

<img width="1440" alt="Screen Shot 2019-09-17 at 4 59 46 PM" src="https://user-images.githubusercontent.com/9412986/65038727-3c21f880-d96e-11e9-82d0-5e4dacbd15e5.png">

rajrishav0711 committed:
Track return tax refund correctly using return_request's request_items

ashishtajane commented on app/models/solidus_avatax_certified/line.rb:
```diff
@@ -75,22 +75,15 @@ def shipment_line(shipment)
-
-        lines << return_item_line(inv_unit.first.line_item, quantity, amount)
+      return_request = @refund.reimbursement.return_items.first.return_authorization.
+        return_requests.first
```
Do we need to use safe access anywhere here?

ashishtajane replied:
What if customer did not return all the items they were supposed to return? (and we issued less refund) -> won't this record more tax refund than what was issued?

rajrishav0711 committed:
Track return tax refund using actual_return_items

if actual_return_items are not created when refund is done,
we fall back to return_request request_items
